tests/ui/lint/lint-ctypes-cstr.rs

```diff
@@ -0,0 +1,16 @@
+#![crate_type = "lib"]
+#![deny(improper_ctypes)]
+
+use std::ffi::{CStr, CString};
+
+extern "C" {
+    fn take_cstr(s: CStr);
```

Comment:
Should there be a `&CStr` variant for this test too? (since that's what users are more likely to use, given that `CStr` is `!Sized`).